src/main/java/io/cdap/plugin/gcp/speech/SpeechToTextTransform.java

```diff
@@ -314,49 +325,78 @@ public String getLanguage() {
     }
 
     @Nullable
-    public Integer getSampleRate() {
+    public Integer getSampleRate() throws IllegalArgumentException {
       if (containsMacro("sampleRate")) {
         return null;
       }
+      try {
+        Integer.parseInt(sampleRate);
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException("Sample rate should be a valid number");
+      }
       return Integer.parseInt(sampleRate);
     }
 
-    private void validate(@Nullable Schema inputSchema) {
+    private void validate(@Nullable Schema inputSchema, FailureCollector collector) {
       if (inputSchema != null) {
         // If a inputSchema and audioField is available then verify that the schema does contain the given audioField
         // and that the type is byte array. This will allow to fail fast i.e. during deployment time.
         String audioFieldName = getAudioField();
         if (audioFieldName != null && inputSchema.getField(audioFieldName) == null) {
-          throw new IllegalArgumentException(String.format("Field %s does not exists in the input schema.",
-                                                           audioFieldName));
-        }
-        Schema.Field field = inputSchema.getField(audioFieldName);
-        Schema fieldSchema = field.getSchema();
-        Schema.Type fieldType =
-          fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
-        if (fieldType != Schema.Type.BYTES) {
-          throw new IllegalArgumentException(String.format("Field '%s' must be of type 'bytes' but is of type " +
-                                                             "'%s'.", field.getName(), fieldType));
+          collector.addFailure(
+            String.format("Field '%s' does not exists in the input schema.", audioFieldName),
+            String.format("Make sure field '%s' exists in the input schema.", audioFieldName))
+            .withConfigProperty(NAME_AUDIOFIELD);
+        } else {
+          Schema.Field field = inputSchema.getField(audioFieldName);
+          Schema fieldSchema = field.getSchema();
+          Schema.Type fieldType =
+            fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
+          if (fieldType != Schema.Type.BYTES) {
+            collector.addFailure(
+              String.format("Field '%s' must be of type 'bytes' but is of type '%s'.", field.getName(), fieldType),
```

Comment:
one thing we've run into several times with the error message is that there isn't an easy way to get a user friendly string for the schema. Here, it will use the physical type, but it's better to use the logical type if the schema has one. We should consider adding a method to the Schema class, or adding a Schemas utility class to hydrator-common for getting a display name or something.

Also, if the physical type is bytes but the logical type is decimal, we should still fail.

Reply:
yea, I agree. Filed https://issues.cask.co/browse/CDAP-15788